### ISSUE TITLE
Presence of bidirectional unicode characters in file

### ISSUE THREAD

**Issue by jbrry** [UniversalDependencies/UD_Irish-TwittIrish#2]
In the file `ga_twittirish-ud-dev.conllu` the following sentence contains 'bidirectional Unicode text' in row 22:
```
# sent_id = NTC_879
# text = RT @user1597: Is rud amháin é bheith I d'Éireannach , ach is rud eile é bheith I do Ghael  #happysaintpatricksday
1       RT      RT      SYM     _       _       5       parataxis:rt    _       _
2       @user1597       @user1597       PROPN   _       _       5       vocative:mention        _       SpaceAfter=No
3       :       :       PUNCT   _       _       5       punct   _       _
4       Is      is      AUX     _       _       5       cop     _       Lang=ga
5       rud     rud     NOUN    _       _       0       root    _       Lang=ga
6       amháin  amháin  ADJ     _       _       5       amod    _       Lang=ga
7       é       é       PRON    _       _       8       nmod    _       Lang=ga
8       bheith  bheith  NOUN    _       _       5       csubj:cop       _       Lang=ga
9       I       i       ADP     _       _       11      case    _       CorrectForm=i|Lang=ga
10      d'      do      DET     _       _       11      det:poss        _       SpaceAfter=No
11      Éireannach      Éireannach      PROPN   _       _       8       xcomp:pred      _       Lang=ga
12      ,       ,       PUNCT   _       _       15      punct   _       _
13      ach     ach     CCONJ   _       _       15      cc      _       Lang=ga
14      is      is      AUX     _       _       15      cop     _       Lang=ga
15      rud     rud     NOUN    _       _       5       conj    _       Lang=ga
16      eile    eile    DET     _       _       15      det     _       Lang=ga
17      é       é       PRON    _       _       18      nmod    _       Lang=ga
18      bheith  bheith  NOUN    _       _       15      csubj:cop       _       Lang=ga
19      I       i       ADP     _       _       21      case    _       CorrectForm=i|Lang=ga
20      do      do      ADP     _       _       21      case    _       Lang=ga
21      Ghael   gael    PROPN   _       _       18      xcomp:pred      _       Lang=ga
22                    SYM     _       _       5       discourse:emo   _       _
23      #happysaintpatricksday  #happysaintpatricksday  X       _       _       5       parataxis:hashtag       _       Lang=en
```

My python 3.9 environment parses the token as '\ue110' but the token seems to have non-deterministic behaviour (prints to different characters between different environments):

```
(Pdb) print('\ue110')
print('\ue110')

```

I was wondering would there be a way to investigate what this token should be and perhaps normalise it to standard Unicode text?

cc @laurenCassidy @tlynn747

**Comment by tlynn747** [comment]
Lauren's original data dump might have it ...

but a quick Twitter (X) search shows this user's bio with the same line:
https://twitter.com/NiamhCranney

The symbol is a star in that instance. But 19 is a lower case i and Ghael has lower case g! So probably not the same string :)
If it's not a star it's likely to be some emoji/pictogram...

**Comment by kscanne** [comment]
Original source is this tweet, I believe:

https://twitter.com/GlorAnTuaiscirt/status/313260061852053505

And yes there is an embedded U+E110.  This character is in the "private use area" of Unicode, and therefore its behavior is not specified by the standard. Hard to know what was intended, although a platform-specific emoji of some kind seems likely given the context.

**Comment by jbrry** [comment]
Thanks for looking into this! For context, I noticed this token would "vanish" after being tokenized/detokenized with either `bert-base-multilingual-cased` or `xlm-roberta-base` and would cause an `IndexError` later on in the code.

The below is a minimal reproducible example:

```
import torch
from transformers import AutoTokenizer, AutoModelForMaskedLM

text = "RT I do Ghael \ue110 test a b c"
model_name = "bert-base-multilingual-cased"
#model_name = "xlm-roberta-base"

tokenizer = AutoTokenizer.from_pretrained(model_name)
model = AutoModelForMaskedLM.from_pretrained(model_name)

inputs = tokenizer.encode(text)
detok = tokenizer.decode(inputs, skip_special_tokens=True)

print(f"{text}, length={len(text.split())}")
print(f"{detok}, length={len(detok.split())}")

-> RT I do Ghael  test a b c, length=9
-> RT I do Ghael test a b c, length=8
```

Even though it's a tokenizer-related issue, it's still probably worth looking at given that a lot of people would be using similar tokenizers.

**Comment by robvanderg** [comment]
I don't think we (you?) should adapt our input data to make it work with certain models. This is a test for robustness that I think should be kept; if you have a model that you want to use for social media data, it should be able to do just that. (Just my opinion)

**Comment by jbrry** [comment]
That's a very valid point. As long as the character is not breaking any UD-specific regulations then it should be up to us as users of the data to work around it. Happy to hear what others think!

**Comment by robvanderg** [comment]
FYI: I have written a script that maps the original text back to the subword ID's, then one can at least run normal evaluation scripts afterwards. Note that this sounds trivial, but is quite tricky to get correct. If this is of interest, you can contact me for copy.

**Comment by jbrry** [comment]
I agree with Rob that if this token is representative of social media text, then there is no need to modify it. Thanks everyone for your help.
